src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +526,201 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
+    from .util import AUTOSCALE_TIMEZONES
+    return AUTOSCALE_TIMEZONES
+
+
+def create_autoscale_condition(cmd, client, resource_group_name, cluster_name, days, time, count):
+    from azure.mgmt.hdinsight.models import AutoscaleRecurrence, AutoscaleSchedule, AutoscaleTimeAndCapacity
+    autoscale_configuration = None
+    cluster = client.get(resource_group_name, cluster_name)
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError(
+            'Please use `az hdinsight autoscale create` to enable autoscale firstly.')
+
+    # try to add schedule condition
+    condition = AutoscaleSchedule(
+        days=days,
+        time_and_capacity=AutoscaleTimeAndCapacity(
+            time=time,
+            min_instance_count=count,
+            max_instance_count=count)
+    )
+    if autoscale_configuration.recurrence:
+        autoscale_configuration.recurrence.schedule.append(condition)
+    elif not autoscale_configuration.recurrence:
+        autoscale_configuration.recurrence = AutoscaleRecurrence(
+            schedule=[condition]
+        )
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def update_autoscale_condition(cmd, client, resource_group_name, cluster_name, index, days=None, time=None, count=None):
+    autoscale_configuration = None
+    cluster = client.get(resource_group_name, cluster_name)
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError('The cluster has not enabled autoscale.')
+
+    if not autoscale_configuration.recurrence:
+        raise CLIError('The cluster does not have any schedule conditions to update.')
+
+    index = int(index)
+    if index >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError(
+            'This cluster only has {} condition(s). Please specify a correct index which starts with 0.'.format(
+                len(autoscale_configuration.recurrence.schedule)))
+
+    # try to update schedule condition
+    if days:
+        autoscale_configuration.recurrence.schedule[index].days = days
+    if time:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.time = time
+    if count:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.min_instance_count = count
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.max_instance_count = count
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def delete_autoscale_condition(cmd, client, resource_group_name, cluster_name, index):
```

Comment:
Need we to handle the exception that index is not int type?

Reply:
monitor doesn't add type=int and doesn't handle the not int excepiton. 
Thanks for your suggestion, I have added type=int, so we don't need to handle this exception in code.

---
In reply to: [466815339](https://github.com/Azure/azure-cli/pull/14692#discussion_r466815339) [](ancestors = 466815339)